packages/eas-cli/src/commandUtils/EasCommand.ts

```diff
@@ -106,37 +115,46 @@ export default abstract class EasCommand extends Command {
   ): Promise<ContextOutput<C>> {
     const contextDefinition = commandClass.contextDefinition;
 
-    const contextValuePairs = await Promise.all(
-      Object.keys(contextDefinition).map(async contextKey => {
-        return [
-          contextKey,
-          await contextDefinition[contextKey].getValueAsync({
-            nonInteractive,
-            sessionManager: this.sessionManager,
-          }),
-        ];
-      })
-    );
+    // do these serially so that they don't do things like ask for login twice in parallel
+    const contextValuePairs = [];
+    for (const [contextKey, contextField] of Object.entries(contextDefinition)) {
+      contextValuePairs.push([
+        contextKey,
+        await contextField.getValueAsync({
+          nonInteractive,
+          sessionManager: this.sessionManager,
+          analyticsManager: this.analyticsManager,
+        }),
+      ]);
+    }
 
     return Object.fromEntries(contextValuePairs);
   }
 
   /**
    * The user session manager. Responsible for coordinating all user session related state.
-   * If needed in a subclass, SessionManager ContextOption.
+   * If needed in a subclass, use the SessionManager ContextOption.
+   */
+  private sessionManager!: SessionManager;
+
+  /**
+   * The analytics manager. Used for logging analytics.
+   * It is set up here to ensure a consistent set up.
    */
-  private readonly sessionManager = new SessionManager();
+  private analyticsManager!: IAnalayticsManagerWithOrchestration;
```

Comment:
In other places we use the following pattern to get rid of `!`:
<img width="408" alt="Screenshot 2022-10-17 at 09 25 49" src="https://user-images.githubusercontent.com/5256730/196114448-96bbc615-d0cb-4840-a08f-dbcd9938516c.png">

Reply:
Had this at first, but `_`-prefixed private members are disallowed in this repo so I went with this. I'll change back and choose a different name.